joint_trajectory_controller/test/config/test_joint_trajectory_controller.yaml

```diff
@@ -10,3 +10,33 @@ test_joint_trajectory_controller:
   state_interfaces:
     - position
     - velocity
+
+  command_joints:
+    - joint5
+    - joint6
+
+  allow_partial_joints_goal: true
+
+  open_loop_control: true
+
+  allow_integration_in_goal_trajectories: true
+
+  action_monitor_rate: 30.0
+
+  interpolation_method: "splines"
+
+  gains:
+    __map_joints:
```

Comment:
`__map_joints` is only a keyword for the GenerateParameterLibrary, you have to create a map for every controlled joint. E.g., see the online docs https://control.ros.org/master/doc/ros2_controllers/joint_trajectory_controller/doc/userdoc.html

Reply:
Thank you @christophfroehlich . I am updating it based on your suggestion. I have a question, Do I need add joints given under "command_joints" too if those joints are different from "joints". 
For example:
joints:
    - joint1
    - joint2
command_joints:
    - joint3
in this case, do i need to create map for joint1,joint2 and joint3?. I am confused between joints and command_joints. I really appreciate if you can explain a bit. Thank you

Reply:
I defined the parameters as follows. 
```
gains:
    __map_joints:
        joint1:
          p: 10.0
          i: 1.0
          d: 1.0
          i_clamp: 10.0
          ff_velocity_scale: 0.0
          normalize_error: 0.02
        joint2:
          p: 10.0
          i: 1.0
          d: 1.0
          i_clamp: 10.0
          ff_velocity_scale: 0.0
          normalize_error: 0.02
```

is it correct format?

Reply:
Not exactly, it works like this way: (and normalize_error is a boolean)

```
gains:
    joint1:
      p: 10.0
      i: 1.0
      d: 1.0
      i_clamp: 10.0
      ff_velocity_scale: 0.0
      normalize_error: false
    joint2:
      p: 10.0
      i: 1.0
      d: 1.0
      i_clamp: 10.0
      ff_velocity_scale: 0.0
      normalize_error: false
```

Reply:
> Thank you @christophfroehlich . I am updating it based on your suggestion. I have a question, Do I need add joints given under "command_joints" too if those joints are different from "joints". For example: joints: - joint1 - joint2 command_joints: - joint3 in this case, do i need to create map for joint1,joint2 and joint3?. I am confused between joints and command_joints. I really appreciate if you can explain a bit. Thank you

have a look at this code here. It has to have the same size.

https://github.com/ros-controls/ros2_controllers/blob/48cc202108d4c91af2556176cae0c796f31d40fa/joint_trajectory_controller/src/joint_trajectory_controller.cpp#L559-L572

I'm not sure actually if the tests use the yaml or hardcode the parameters in every case
https://github.com/ros-controls/ros2_controllers/blob/48cc202108d4c91af2556176cae0c796f31d40fa/joint_trajectory_controller/test/test_trajectory_controller_utils.hpp#L129

If your yaml file does not throw an error: I suggest you investigating this to be sure, starting here:
https://github.com/ros-controls/ros2_controllers/blob/master/joint_trajectory_controller/test/test_trajectory_controller.cpp#L368

Reply:
@christophfroehlich I went through the files you mentioned earlier. I think I have got an Idea of what's going on here. I want to double check with you whether my understand is correct or not. In this particular issue, we need to read and assign the trajectory controller param values from yaml file. We are using generate_param_library to generate header files for this parameters in yaml and we are using these param header files to retrieve and assign the values to trajectory controller parameter. And our test need to verify whether assigned trajectory controller parameters values properly matched with non default values given in the respective yaml file.

And i think you are right, we need to create new test case as you mentioned below. Also, you mentioned that I can use "param_" variable  to retrieve the parameter values using traj_controller_ object. But, I think, First, we have to  assign the values from header file to traj_controller_ using the following function in our test case
https://github.com/ros-controls/ros2_controllers/blob/48cc202108d4c91af2556176cae0c796f31d40fa/joint_trajectory_controller/test/test_trajectory_controller_utils.hpp#L195-L216

I really appreciate it if you can have look and let me whether my understand is right or wrong. If  my approach is right
1) how can i create " std::vector<rclcpp::Parameter> & parameters = {}" argument in the below function  by using yaml parameters header files created by generate param library
https://github.com/ros-controls/ros2_controllers/blob/48cc202108d4c91af2556176cae0c796f31d40fa/joint_trajectory_controller/test/test_trajectory_controller_utils.hpp#L195

2) And also how can i retrieve protected "param_" variable using traj_controller_ object when there is no getter function for that variable.

Thank you and looking forward to your reply. have a good day and sorry for long question :)

Reply:
As @bmagyar has pointed out, we don't have to test if the values are read correctly from generate_parameter_library (we assume that this dependency works as expected), but test if JTC works with non-default parameters. 
Maybe @destogl can comment in the original issue which tests exactly he had in mind.
As a prerequisite for that we have to understand how we can load/use the yaml file in the tests -> I honestly also have to see and play around how this works.

Reply:
Hi @christophfroehlich , I am still struggling to understand what needs to be done exactly in this test and how to retrieve the parameters. I wondering whether you get the chance to have a look at it?